site/src/components/Resources/AgentMetadata.tsx

```diff
@@ -98,16 +98,14 @@ export const AgentMetadataView: FC<AgentMetadataViewProps> = ({ metadata }) => {
     return null;
   }
   return (
-    <div css={styles.root}>
-      <Stack alignItems="baseline" direction="row" spacing={6}>
-        {metadata.map((m) => {
-          if (m.description === undefined) {
-            throw new Error("Metadata item description is undefined");
-          }
-          return <MetadataItem key={m.description.key} item={m} />;
-        })}
-      </Stack>
-    </div>
+    <section css={styles.root}>
+      {metadata.map((m) => {
+        if (m.description === undefined) {
+          throw new Error("Metadata item description is undefined");
```

Comment:
This component checks to see if each piece of metadata has a description. [Above](https://github.com/coder/coder/pull/11647/files#diff-593c71a20bc7ee090cfea5f070f9174b8ab3d28df7f03f5a461a147429d2f04bR97), it checks to see if the metadata array is empty. And the parent component checks to see if [metadata is undefined](https://github.com/coder/coder/pull/11647/files#diff-593c71a20bc7ee090cfea5f070f9174b8ab3d28df7f03f5a461a147429d2f04bR161), while also[ looping over](https://github.com/coder/coder/pull/11647/files#diff-593c71a20bc7ee090cfea5f070f9174b8ab3d28df7f03f5a461a147429d2f04bR171) the metadata array (and looking at the description again).

Perhaps we could tighten up these checks, only displaying the `AgentMetadataView` component if it has length and isn't undefined, and only looping in one place.

Reply:
Those checks are not necessary! Thanks for pointing them out